sqlglot/diff.py

```diff
@@ -199,11 +214,22 @@ def _generate_edit_script(self, matchings: t.Dict[int, int], delta_only: bool) -
             source_node = self._source_index[kept_source_node_id]
             target_node = self._target_index[kept_target_node_id]
 
-            if (
-                not isinstance(source_node, UPDATABLE_EXPRESSION_TYPES)
-                or source_node == target_node
-            ):
-                edit_script.extend(self._generate_move_edits(source_node, target_node, matchings))
+            identical_nodes = source_node == target_node
```

Comment:
I decided to cache the hashes because this equality check is gonna be hit a lot, since `matchings` must be generally quite larger than the unmatched source/target sets.